unstructured/partition/email.py

```diff
@@ -264,12 +301,24 @@ def partition_email(
                 _replace_mime_encodings = partial(replace_mime_encodings, encoding=encoding)
                 try:
                     element.apply(_replace_mime_encodings)
-                except UnicodeDecodeError:
-                    # If decoding fails, try decoding with default encoding (utf-8)
-                    element.apply(replace_mime_encodings)
+                except (UnicodeDecodeError, UnicodeError):
```

Comment:
It might be useful to pull this logic into a function in `encoding.py` in case we need to do something similar in another partitioning function (like `partition_text` or `partition_msg`).

Reply:
@MthwRobinson @cragwolfe There is already similar logic in the `detect_file_encoding` function in `encoding.py` (https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/file_utils/encoding.py#L55-L62). The above logic was added separately to handle Unicode errors from the `replace_mime_encodings` function in `unstructed/cleaners/core.py`. Other partitioning functions (like partition_text or partition_msg) don't seem to use this `replace_mime_encodings` function. 

Do you mean to pull this logic into the `replace_mime_encodings` function in `unstructed/cleaners/core.py`?